lib/openhab/dsl/types/datetime.rb

```diff
@@ -0,0 +1,339 @@
+      # way of breaking it up into multiple classes
+      class DateTime
+        extend Forwardable
+        include Comparable
+        include OpenHAB::Log
+
+        def_delegator :datetime, :to_s
+        def_delegator :zoned_date_time, :month_value, :month
+        def_delegator :zoned_date_time, :minute, :min
+        def_delegator :zoned_date_time, :second, :sec
+        def_delegator :zoned_date_time, :nano, :nsec
+        alias inspect to_s
+
+        java_import org.openhab.core.library.types.DateTimeType
+        java_import java.time.ZonedDateTime
+        java_import java.time.Instant
+        java_import java.time.ZoneId
+        java_import java.time.ZoneOffset
+        java_import java.time.Duration
+
+        #
+        # Regex expression to identify strings defining a time in hours, minutes and optionally seconds
+        #
+        TIME_ONLY_REGEX = /\A\d\d:\d\d(:\d\d)?/.freeze
+
+        attr_reader :datetime
+
+        #
+        # Create a new DateTime instance wrapping an OpenHAB DateTimeType
+        #
+        # @param [Java::org::openhab::core::library::types::DateTimeType] datetime The DateTimeType instance to
+        #   delegate to, or an object that can be converted to a DateTimeType
+        #
+        def initialize(datetime)
+          @datetime = case datetime
+                      when DateTimeType
+                        datetime
+                      when ZonedDateTime
+                        DateTimeType.new(datetime)
+                      else
+                        datetime_type_from(datetime)
+                      end
+        end
+
+        #
+        # Compare thes DateTime object to another
+        #
+        # @param [Object] other Other object to compare against
+        #
+        # @return [Integer] -1, 0 or 1 depending on the outcome
+        #
+        def <=>(other)
+          logger.trace("Comparing self (#{self.class}) to #{other} (#{other.class})")
+          case other
+          when DateTime, DateTimeType, DateTimeItem
+            zoned_date_time.to_instant.compare_to(other.zoned_date_time.to_instant)
+          when TimeOfDay::TimeOfDay, TimeOfDay::TimeOfDayRangeElement
+            to_tod <=> other
+          else
+            self <=> DateTime.new(other)
+          end
+        end
+
+        #
+        # Adds another object to this DateTime
+        #
+        # @param [Object] other Object to add to this. Can be a Numeric, another DateTime/Time/DateTimeType, a
+        #   Duration or a String that can be parsed into a DateTimeType or Time object
+        #
+        # @return [DateTime] A new DateTime object representing the result of the calculation
+        #
+        # rubocop: disable Metrics/MethodLength
+        # Disabled line length - case dispatch pattern
+        def +(other)
+          logger.trace("Adding #{other} (#{other.class}) to #{self}")
+          case other
+          when Numeric
```

Comment:
convert to the one line form of the when statement.

when Number then ...
when DateTime, time then...

Then you don't need to disable the rubocop check above.